samza-core/src/main/java/org/apache/samza/container/grouper/task/BalancingTaskNameGrouper.java

```diff
@@ -54,5 +55,11 @@
    * @param localityManager provides a persisted task to container map to use as a baseline
    * @return                the grouped tasks in the form of ContainerModels
    */
-  Set<ContainerModel> balance(Set<TaskModel> tasks, LocalityManager localityManager);
+  default Set<ContainerModel> balance(Set<TaskModel> tasks, LocalityManager localityManager) {
```

Comment:
Wait, this seems not making much sense to me if this interface has only one api method and it's a default. If existing classes doesn't want to implement the single api method, why bother implementing this interface? Seems pretty confusing to me.